OsmAnd-java/src/main/java/net/osmand/GPXUtilities.java

```diff
@@ -1809,6 +1814,37 @@ public String getRef() {
 			return null;
 		}
 
+		public String getOuterRadius() {
+			String outerRadius = "";
+			QuadRect rect = getRect();
+			if (rect.width() > 0.00001 && rect.height() > 0.00001) {
+				WptPt wptPt = findPointToShow();
+				if (wptPt != null) {
+					int radius = (int) Math.max(MapUtils.getDistance(wptPt.lat, wptPt.lon, rect.bottom, rect.left),
+							MapUtils.getDistance(wptPt.lat, wptPt.lon, rect.top, rect.right));
+					outerRadius = getRadiusID(radius);
+				}
+			}
+			return outerRadius;
+		}
+
+		/**
+		 * convert radius to char to store in the obf file
+		 *
+		 * @param radius integer radius in meters
+		 * @return String where  A <= 5 km, B <= 10 km, C <= 50 km, D <= 100 km, E <= 500 km, F <= 1000 km,
+		 * G <= 5000 km, H <= 10000 km
+		 */
+		private String getRadiusID(int radius) {
```

Comment:
this method should be within MapUtils as reverse method as well. Parameters could be 'A' - letter, 5000 - start, [2, 5]  - seq